### PR TITLE
Adding additional fade types to single strand effect and tool tips.

### PR DESCRIPTION
**Pull request by sidelights**
changed 3d fade checkbox to choice, to include 'From head' old 3d fade, from tail reverse of 3d fade, head and tail bright at the ends, and middle bright in the middle. Also added tool tip text for Single Strand panel.

### CODE REVIEW

File: xLights/effects/SingleStrandEffect.cpp [hunk 1]
```diff
@@ -661,13 +685,99 @@ void SingleStrandEffect::draw_chase(RenderBuffer &buffer,
...
+                        color = hsv1;
+                    }
+                } else if (Fade_Type == "From Tail") {
+                    if (buffer.allowAlpha) {
+                        // reverse the fade black to white
+                        color.alpha = 255.0 * (max_chase_width - i + 1.0) / max_chase_width;
+                    } else {
+                        HSVValue hsv1 = color.asHSV();
+                        // new reverse value (brightnesss fade) take the orignal brightness - itself (min value) + the fraction of the width going to bright
+                        hsv1.value = orig_v - orig_v + ((max_chase_width - (i + 1.0)) / max_chase_width);
```
Comment (derwin12) [inline xLights/effects/SingleStrandEffect.cpp:707]: interesting math there..

File: xLights/effects/SingleStrandEffect.cpp [hunk 2]
```diff
@@ -661,13 +685,99 @@ void SingleStrandEffect::draw_chase(RenderBuffer &buffer,
...
+                        if (i == i ) {
+                            color.alpha = color.alpha;
+                        }
+
+                    } else {   // head tail not alpha 
+                        HSVValue hsv1 = color.asHSV();
+                            if (i > middle_chase_index) {
+                                hsv1.value = orig_v - ((max_chase_width - (2.0 * i + 1.0)) / max_chase_width);
+                            } else {
+                                hsv1.value = orig_v - orig_v + ((max_chase_width - (2.0 * i )) / ( max_chase_width));
```
Comment (derwin12) [inline xLights/effects/SingleStrandEffect.cpp:739]: same here

File: xLights/effects/SingleStrandEffect.cpp [hunk 3]
```diff
@@ -661,13 +685,99 @@ void SingleStrandEffect::draw_chase(RenderBuffer &buffer,
...
+                            }
+                        }
+                        if (i == i) {
+                            color.alpha = color.alpha;
+                        }
+
+                    } else {  // middle not alpha blend
+                        HSVValue hsv1 = color.asHSV();
+                        if (i > middle_chase_index) {
+                            hsv1.value = orig_v - orig_v + ((max_chase_width - (i + 1.0)) / (.5 * max_chase_width));
```
Comment (derwin12) [inline xLights/effects/SingleStrandEffect.cpp:773]: ditto

File: xLights/effects/SingleStrandEffect.cpp (hunk 1 above)
Comment (sidelights) [reply xLights/effects/SingleStrandEffect.cpp:707]: removed effective 0's

File: xLights/effects/SingleStrandEffect.cpp (hunk 2 above)
Comment (sidelights) [reply xLights/effects/SingleStrandEffect.cpp:739]: removed effective 0's

File: xLights/effects/SingleStrandEffect.cpp (hunk 3 above)
Comment (sidelights) [reply xLights/effects/SingleStrandEffect.cpp:773]: removed effective 0's

**Pull request merged**